x-pack/platform/plugins/shared/fleet/server/services/package_policies/utils.test.ts

```diff
@@ -253,3 +254,144 @@ describe('canDeployAsAgentlessOrThrow', () => {
     expect(error).toBe(null);
   });
 });
+
+describe('removeCloudConnectorTransientVars', () => {
+  it('should remove cloud_connector_name transient var from all streams', () => {
+    const packagePolicy = {
+      name: 'test-policy',
+      inputs: [
+        {
+          type: 'test-input',
+          streams: [
+            {
+              id: 'stream-1',
+              vars: {
+                cloud_connector_name: { value: 'my-connector', type: 'text' },
```

Comment:
We need to avoid storing the `cloud_connector_name` as input var. It should stored as part of Cloud connector save object. Similar Package Policy has text field but name is store as part of package policy save object, not a input var field.